source/simulator/core.cc

```diff
@@ -2241,6 +2256,297 @@ namespace aspect
+                  current_linearization_point.block(introspection.block_indices.velocities) += search_direction.block(introspection.block_indices.velocities);
+
+                  assemble_newton_stokes_matrix = rebuild_stokes_preconditioner = false;
+                  rebuild_stokes_matrix = true;
+
+                  assemble_stokes_system();
+
+                  test_newton_residual_velo = system_rhs.block(introspection.block_indices.velocities).l2_norm();
+                  test_newton_residual_pres = system_rhs.block(introspection.block_indices.pressure).l2_norm();
+                  test_newton_residual = std::sqrt(test_newton_residual_velo * test_newton_residual_velo + test_newton_residual_pres * test_newton_residual_pres);
+
+                  if (test_newton_residual < (1.0 - alfa * lambda) * newton_residual || line_search_iteration >= parameters.max_newton_line_search_iterations || use_picard)
+                    {
+                      pcout << newton_residual/initial_newton_residual << "   Norm of rhs = " << newton_residual << ":" << test_newton_residual  << ", relative residual: " << newton_residual/initial_newton_residual << ", theta = " << parameters.newton_theta << std::endl;
+                      break;
+                    }
+                  else
+                    {
+
+                      pcout << "   Iteration " << line_search_iteration << ", with norm of the rhs " << test_newton_residual << " and going to "
+                            << (1.0 - alfa * lambda) * newton_residual << ", relative residual: " << newton_residual/initial_newton_residual << ", theta = " << parameters.newton_theta << std::endl;
+
+                      lambda = lambda * (2.0/3.0);// TODO: make a parameter out of this.
+                    }
+
+                  line_search_iteration++;
+                }
+              while (true);
+
+
+              if (use_picard == true)
+                {
+                  parameters.switch_initial_newton_residual = newton_residual;
+                  parameters.newton_residual = newton_residual;
+                }
+              else
+                {
+                  bool use_said_method = false;
+                  if (use_said_method) //said's method
+                    parameters.newton_residual = test_newton_residual;
+                  else
+                    parameters.newton_residual = 0;
+                }
+
+              newton_residual_old = newton_residual;
+
+              pcout << std::endl;
+              this->last_pressure_normalization_adjustment = normalize_pressure(current_linearization_point);
+              ++nonlinear_iteration;
+
+              if (newton_residual/initial_newton_residual < parameters.nonlinear_tolerance)
+                break;
+            }
+
+          pcout << "Final residual: Norm of rhs = " << newton_residual << ", relative residual: " << newton_residual/initial_newton_residual <<  std::endl;
```

Comment:
Please adjust the style of this output to be similar to the other solver schemes.